### PR TITLE
Add hide_title metadata that hides the title text on the top of the doc

### PR DESCRIPTION
**Pull request by MisterTea**
## Motivation

Optionally hide the title from the top of the doc so people can customize the title

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Running with this on my test site

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

**Comment by yangshun** [comment]
@MisterTea Did you run prettier before submitting this PR?

File: lib/core/DocsLayout.js [hunk 1]
```diff
@@ -23,6 +23,15 @@ class DocsLayout extends React.Component {
     if (this.props.Doc) {
       DocComponent = this.props.Doc;
     }
+    let docTitle = i18n
```
Comment (JoelMarcey) [inline lib/core/DocsLayout.js]: Someone might put 

```
hide_title: false
```

in the doc header.

So, how about instead we do the converse here:

```
let docTitle = null;
// If the attribute does not exist or is set to false, we set a title.
if (!this.props.metadata.hide_title || this.props.metadata.hide_title.toLowerCase() === 'false') {
  docTitle = ....
}
```

File: lib/core/Doc.js [hunk 2]
```diff
@@ -60,11 +60,18 @@ class Doc extends React.Component {
         </a>
       );
     }
+
+    let title = null;
+    if (this.props.title) {
```
Comment (JoelMarcey) [inline lib/core/Doc.js]: Maybe a comment here saying something like:

```
// this.props.title could be set to null if hide_title was set in the doc metadata
```

**Comment by JoelMarcey** [comment]
@yangshun Can you look at this too? I think it is right, but just wanted to make sure I wasn't missing anything here.

File: docs/api-doc-markdown.md [hunk 3]
```diff
@@ -13,6 +13,8 @@ Documents use the following markdown header fields that are enclosed by a line `
 
 `title`: The title of your document. If this field is not present, the document's `title` will default to its `id`.
 
+`hide_title`: Whether to hide the title at the top of each documentation page. If this is `null` or set to anything besides `false`, the title will be displayed.
```
[the remark this answers is not in the document]
Comment (JoelMarcey) [reply docs/api-doc-markdown.md]: @yangshun How do you control what a user might enter in the metadata for the doc though? Someone could put:

```
hide_title: 'blahblah'
```

How do we treat that?

Or are you saying that just the fact that `hide_title` is explicitly put in the metadata it is always truthy, no matter the associated string?

**Comment by MisterTea** [comment]
I think this new way is a lot cleaner, thanks.  Let me know what you think!

File: lib/core/Doc.js [hunk 4]
```diff
@@ -60,11 +60,12 @@ class Doc extends React.Component {
         </a>
       );
     }
+
     return (
       <div className="post">
         <header className="postHeader">
           {editLink}
-          <h1>{this.props.title}</h1>
+          {this.props.hideTitle && <h1>{this.props.title}</h1>}
```
Comment (JoelMarcey) [inline lib/core/Doc.js]: Wait. Shouldn't this be 

```
!this.props.hideTitle
```

We only want to show this if we don't have hideTitle, right?

**Comment by yangshun** [comment]
Good catch! My bad for typing incorrect example code >_<

**Comment by JoelMarcey** [comment]
I went ahead and made the update. I think we are good now.

**Comment by yangshun** [comment]
I'll give it a thorough test later (:

On Apr 11, 2018 5:53 PM, "Joel Marcey" <notifications@github.com> wrote:

*@JoelMarcey* approved this pull request.

—

You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/facebook/Docusaurus/pull/540#pullrequestreview-111450160>,
or mute the thread

**Comment by JoelMarcey** [comment]
I was going to test it locally right now ;)

**Comment by JoelMarcey** [comment]
Seems to work 👍

**Pull request merged**